src/wazuh_modules/syscollector/sysOsParsers.cpp

```diff
@@ -0,0 +1,402 @@
+/*
+ * Wazuh SysInfo
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * October 28, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "sysOsParsers.h"
+#include "stringHelper.h"
+#include <regex>
+
+static bool parseUnixFile(const std::map<std::string, std::string>& keyMap,
+                          const char separator,
+                          std::istream& in,
+                          nlohmann::json& info)
+{
+    enum ValueIds
+    {
+        KEY_ID,
+        DATA_ID,
+        MAX_ID
+    };
+    bool ret{false};
+    std::string line;
+    while(std::getline(in, line))
+    {
+        line = Utils::trim(line);
+        const auto data{Utils::split(line, separator)};
+        if (data.size() == MAX_ID)
+        {
+            const auto it
+            {
+                std::find_if(keyMap.cbegin(), keyMap.cend(),
+                [&data](const auto& value)
+                {
+                    return value.first == Utils::trim(data[KEY_ID]);
+                })
+            };
+            if (it != keyMap.cend())
+            {
+                info[it->second] = Utils::trim(data[DATA_ID], " \"\t");
+                ret = true;
+            }
+        }
+    }
+    return ret;
+}
+
+
+
+static bool findRegexInString(const std::string& in,
+                              std::string& match,
+                              const std::regex& pattern,
+                              const size_t matchIndex = 0,
+                              const std::string& start = "")
+{
+    bool ret{false};
+    if (start.empty() || Utils::startsWith(in, start))
+    {
+        std::smatch sm;
+        ret = std::regex_search(in, sm, pattern);
+        if (ret && sm.size() >= matchIndex)
+        {
+            match = sm[matchIndex];
+        }
+    }
+    return ret;
+}
+
+static bool findCodeNameInString(const std::string& in,
+                                 std::string& output)
+{
+    const auto end{in.rfind(")")};
+    const auto start{in.rfind("(")};
+    const bool ret
+    {
+        start != std::string::npos && end != std::string::npos
+    };
+    if (ret)
+    {
+        output = Utils::trim(in.substr(start + 1, end - (start + 1)));
+    }
+    return ret;
+}
+
+static bool findMajorMinorVersionInString(const std::string& in,
+                                         nlohmann::json& output)
+{
+    constexpr auto MINOR_VERSION_PATTERN{"^[0-9]+\\.([0-9]+)\\.*"};
+    constexpr auto MAJOR_VERSION_PATTERN{"^([0-9]+)\\.*"};
+    std::string version;
+    std::regex pattern{MAJOR_VERSION_PATTERN};
+    const auto ret{findRegexInString(in, version, pattern, 1)};
+    if (ret)
+    {
+        output["os_major"] = version;
+    }
+    pattern = MINOR_VERSION_PATTERN;
+    if (findRegexInString(in, version, pattern, 1))
```

Comment:
if findRegexInString returns `false` here but, it returns `true` the first time for major version (line 97), the function will return `true` as a final result even when the `output` json has only one value: "os_major". Is that the expected behavior?

Reply:
yes, that's the expected behavior. Some distros doesn't have a minor version like Fedora 22.